### PR TITLE
Add csproj and bash script for crossgen2 ThunkGenerator

### PR DESCRIPTION
**Pull request by jakobbotsch**
I have had to run this from Linux twice and both times I have had to create a csproj first because even using the `csc` that comes with the SDK does not automatically add the needed references.

PTAL @MichalStrehovsky

### CODE REVIEW

File: src/tools/crossgen2/Common/JitInterface/ThunkGenerator/gen.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+cd "$(dirname ${BASH_SOURCE[0]})"
```
Comment (MichalStrehovsky) [inline src/tools/crossgen2/Common/JitInterface/ThunkGenerator/gen.sh]: `cd /d %~dp0` is the equivalent Windows batch command. Could you add that to gen.bat?

Comment (jakobbotsch) [reply src/tools/crossgen2/Common/JitInterface/ThunkGenerator/gen.sh]: Done!

**Comment by jashook** [comment]
@MichalStrehovsky is this good to go?

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky is this good to go?

Yup, merged. Sorry for the delay.

**Pull request merged**